sdks/python/apache_beam/pipeline.py

```diff
@@ -1285,7 +1285,8 @@ def named_inputs(self):
           self.main_inputs, self.side_inputs)
       if not self.parts:
         for name, pc_out in self.outputs.items():
-          if pc_out.producer is not self:
+          if pc_out.producer is not self and pc_out not in named_inputs.values(
```

Comment:
Is this a bug fix that might be best served as a separate commit? Etc elsewhere

Reply:
This resulted in some spurious edges that didn't impact anything but rendering. (The other change was just a refactoring to re-use the side inputs code.)